vault/plugin_catalog.go

```diff
@@ -234,7 +234,7 @@ func (c *PluginCatalog) newPluginClient(ctx context.Context, pluginRunner *plugi
 		id:     id,
 		logger: c.logger.Named(pluginRunner.Name),
 		cleanupFunc: func() error {
-			return c.removePluginClient(pluginRunner.Name, id)
```

Comment:
The context here is likely going to be expired by the time this cleanup function runs

Reply:
That's a good point. I was gonna suggest making `cleanupFunc` context-aware but `Close` doesn't accept a context so it's of no use. Would a background context with some safe timeout value be okay here?

Reply:
If I update `pluginClient` to also store the multiplexing flag then we shouldn't need to pass the context as mentioned [here](https://github.com/hashicorp/vault/pull/14033/files#r806344620).